### PR TITLE
send log traces to application insights

### PR DESCRIPTION
**Pull request by monikavar**

### CODE REVIEW

File: deploy/demo/template.json [hunk 1]
```diff
@@ -363,7 +363,61 @@
              "metadata": {
              "description": "Specifies the location in which to create the workspace."
              }
-        }
+        },
+        "appInsightsName": {
+                "type": "string",
```
Comment (marcschier) [inline deploy/demo/template.json]: Spaces

File: deploy/demo/template.json [hunk 2]
```diff
@@ -951,6 +1040,14 @@
...
             "type": "string",
             "value": "[resourceGroup().name]"
-        }       
+        },
+        "appinsights-name": {
+            "type": "string",
+            "value": "[parameters('appInsightsName')]"
+        },
+        "appinsights-instrumentationkey": {
+        "type": "string",
```
Comment (marcschier) [inline deploy/demo/template.json]: Spaces

File: deploy/demo/template.json [hunk 3]
```diff
@@ -951,6 +1040,14 @@
...
+        "value": "[reference(resourceId('Microsoft.Insights/components', parameters('appInsightsName')), '2014-04-01').InstrumentationKey]"
```
Comment (marcschier) [inline deploy/demo/template.json]: Make api version a variable like we do for the rest.

File: src/Microsoft.Azure.IIoT.Services.OpcUa.Onboarding/src/Program.cs [hunk 4]
```diff
@@ -94,13 +94,14 @@ public class Program {
...
             var builder = new ContainerBuilder();
-
+            
             // Register configuration interfaces
             builder.RegisterInstance(config)
                 .AsImplementedInterfaces().SingleInstance();
 
             // register logger
-            builder.RegisterLogger(LogEx.Console());
+            var logger = new LoggerConfiguration().ApplicationInsights(configuration).CreateLogger();
```
Comment (marcschier) [inline src/Microsoft.Azure.IIoT.Services.OpcUa.Onboarding/src/Program.cs]: Add another method to LogEx, same as the Console() and just rename Console to the new method name.

**Review by marcschier — CHANGES_REQUESTED** [review]

File: deploy/demo/template.json [hunk 5]
```diff
@@ -363,6 +363,60 @@
...
+             "defaultValue": "[concat('appinsights-', take(uniqueString(subscription().subscriptionId, resourceGroup().id, parameters('solutionName')), 5))]",
+             "metadata": {
+                  "description": "Specifies the name of application insights."
+             }
+        },
+        "appInsightsType": {
+             "type": "string",
+             "defaultValue": "web",
+             "allowedValues": [
+                    "web",
```
Comment (marcschier) [inline deploy/demo/template.json]: Is this even needed?   What would be the outcome of changing this?

File: deploy/vm/template.json [hunk 6]
```diff
@@ -951,6 +1041,14 @@
         "resourceGroup" : {
             "type": "string",
             "value": "[resourceGroup().name]"
-        }       
+        },
+        "appinsights-name": {
```
Comment (marcschier) [inline deploy/vm/template.json:1032]: You need to also update the setup.sh file (as well as calling it) with the instrumentation key and id.   The setup.sh is called to setup the vm, as part of the arm template.   It is in the same folder.

**Review by marcschier — CHANGES_REQUESTED** [review]
Looks good - only missing is the change to setup.sh and corresponding update to the invocation from arm template for both vm and demo deployments.

File: deploy/vm/template.json [hunk 7]
```diff
@@ -670,6 +670,49 @@
...
+            "type": "microsoft.insights/components/CurrentBillingFeatures",
+            "location": "[variables('location')]",
+            "apiVersion": "2015-05-01",
+            "dependsOn": [
+                 "[resourceId('microsoft.insights/components', parameters('appInsightsName'))]"
+            ],
+            "properties": {
+                    "CurrentBillingFeatures": "[variables('appInsightsPricePlan')]",
+                    "DataVolumeCap": {
+                        "Cap": "[parameters('appInsightsDailyQuota')]",
```
Comment (marcschier) [inline deploy/vm/template.json:697]: Please use 4 spaces per tab.

**Review by marcschier — CHANGES_REQUESTED** [review]

File: deploy/demo/template.json (hunk 5 above)
Comment (monikavar) [reply deploy/demo/template.json]: Removed it, tested without this and it works

**Review by marcschier — APPROVED** [review]
Please make sure you tested deployment of vm/demo since we do not have CI tests for it yet.

**Pull request merged**